### PR TITLE
[WIP][Refactor] ForumThreadView uses Twig

### LINKED ISSUE #3742: [Refactor] Twig engine for discussion forum
**What problem are you trying to solve with Submitty**
Currently, the Discussion Forum renders HTML directly from the View file. This has to be shifted to be using the Twig template engine for consistency with the rest of the Submitty project. 

**Describe the way you'd like to solve this problem**
Refactor the `app/views/forum/ForumThreadView.php` to use the Twig template engine and put the respective templates in the `app/templates/forum` directory.

### PR DESCRIPTION
**Pull request by xprilion**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Closes #3742 

### What is the new behavior?
The following use Twig templates - 

1. [x] CreateThread
2. [ ] SearchResult
3. [ ] ShowForumThreads
4. [ ] GeneratePostList
5. [ ] DisplayThreadList
6. [ ] CreatePost
7. [ ] StatPage

### Other information?
Is this a breaking change? - None discovered yet.
How did you test - Not yet tested, only proper working checked manually.

### CODE REVIEW

File: site/app/templates/forum/createThread.twig [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{% for css in page_css %}
+    <link rel="stylesheet" href="{{ base_url }}/{{ css }}" />
+{% endfor %}
+
+{% for js in page_js %}
+    <script type="text/javascript" language="javascript" src="{{ base_url }}/{{ js }}"></script>
+{% endfor %}
```
Comment (MasterOdin) [inline site/app/templates/forum/createThread.twig]: Don't inject JS/CSS files directly (I know this was how they were before, and that's a bug) as we end up in a mess with how browsers are caching these files and how we might try and invalidate that cache. It's why there's the `Output::addInternal{Js,Css}` and `Output::addVendor{Js,Css}` methods which add a datestring to the end of the file to help properly invalidate a file on update and force a re-cache.

File: site/app/views/forum/ForumThreadView.php [hunk 2]
```diff
@@ -997,97 +997,60 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
...
-								"thread_exists" => $thread_exists,
-								"show_more" => false
-	]);
+        $page_js = [];
+        $page_js[] = "vendor/jquery.are-you-sure/jquery.are-you-sure.js";
+        $page_js[] = "js/drag-and-drop.js";
+        $page_js = "js/flatpickr.js";
 
+        $page_css = [];
+        $page_css[] = "css/flatpickr.min.css";
```
Comment (MasterOdin) [inline site/app/views/forum/ForumThreadView.php]: Looks like these got missed when I vendored these files. This should be `vendor/flatpickr/flatpickr.min.css` (but use the `Output::addVendorCss` function as explained above). See the rest of the file for examples.

File: site/app/views/forum/ForumThreadView.php [hunk 3]
```diff
@@ -997,97 +997,60 @@ public function createPost($thread_id, $post, $unviewed_posts, $function_date, $
...
-								"forum_bar_buttons_right" => $buttons,
-								"forum_bar_buttons_left" => [],
-								"show_threads" => false,
-								"thread_exists" => $thread_exists,
-								"show_more" => false
-	]);
+        $page_js = [];
+        $page_js[] = "vendor/jquery.are-you-sure/jquery.are-you-sure.js";
+        $page_js[] = "js/drag-and-drop.js";
+        $page_js = "js/flatpickr.js";
```
Comment (MasterOdin) [inline site/app/views/forum/ForumThreadView.php]: Looks like these got missed when I vendored these files. This should be `vendor/flatpickr/flatpickr.min.js` (but use the `Output::addVendorJs` function as explained above). See the rest of the file for examples.

**Comment by xprilion** [comment]
Will open a new PR for this on Submitty repo branch instead of my fork with the suggested changes from @MasterOdin and others discussed in meeting today.

**Pull request closed without merging**